sdk/core/core/inc/az_http_request_builder.h

```diff
@@ -54,7 +68,10 @@ az_result az_http_request_builder_mark_retry_headers_start(az_http_request_build
 
 az_result az_http_request_builder_remove_retry_headers(az_http_request_builder * const p_hrb);
 
-az_result az_http_request_builder_remove_retry_headers(az_http_request_builder * const p_hrb);
+az_result az_http_request_builder_get_header(
```

Comment:
We need some doc for all the methods. What does this return if the index is out of range?

Reply:
I should add doxygen comments for all the functions. (This one will return AZ_ERROR_ARG).

Reply:
I'll add doxygen comments in the follow up PR.